File: data/code_pairs/fortran/70.f90
! Note: This Fortran code assumes the use of a module or interface to bind cuBLAS functions.
! This example illustrates the structure and calls but omits specific cuBLAS bindings for brevity.

module mask_cublas
    use iso_c_binding
    implicit none
    contains

    subroutine cublas_setmat(Msize, matrix, dev_ptr)
        integer, intent(in) :: Msize
        real(kind=8), intent(in) :: matrix(Msize, Msize)
        type(c_ptr), intent(out) :: dev_ptr
        ! Your cuBLAS bindings and initialization here
        print *, "Setting matrix (actual cuBLAS implementation)."
    end subroutine

    subroutine cublas_release(dev_ptr)
        type(c_ptr), intent(inout) :: dev_ptr
        ! Your cuBLAS resource release here
        print *, "Releasing resources (actual cuBLAS implementation)."
    end subroutine

end module mask_cublas

program test_cublas
    use mask_cublas
    implicit none
    integer, parameter :: Msize = 2
    real(kind=8), dimension(Msize, Msize) :: matrix
    type(c_ptr) :: dev_ptr

    matrix = reshape([1.0d0, 2.0d0, 3.0d0, 4.0d0], shape(matrix))

    call cublas_setmat(Msize, matrix, dev_ptr)
    call cublas_release(dev_ptr)

    print *, "Fortran test completed successfully."
end program test_cublas

File: data/code_pairs/cpp/70.cpp
#include <iostream>
#include <cublas_v2.h>
#include <cuda_runtime.h>

class MaskCuBLAS {
public:
    MaskCuBLAS() : cublas_not_init(true) {
        initialize();
    }

    ~MaskCuBLAS() {
        cleanup();
    }

    void cublas_setmat(int Msize, const double* matrix, double** dev_ptr) {
        std::cout << "Setting matrix (actual cuBLAS implementation)." << std::endl;
        cudaMalloc((void**)dev_ptr, sizeof(double) * Msize * Msize);
        cublasSetMatrix(Msize, Msize, sizeof(double), matrix, Msize, *dev_ptr, Msize);
    }

    void cublas_release(double* dev_ptr = nullptr) {
        std::cout << "Releasing resources (actual cuBLAS implementation)." << std::endl;
        if (dev_ptr != nullptr) {
            cudaFree(dev_ptr);
        }
    }

private:
    bool cublas_not_init;
    cublasHandle_t handle;

    void initialize() {
        cublasCreate(&handle);
        cublas_not_init = false;
    }

    void cleanup() {
        cublasDestroy(handle);
    }
};

int main() {
    MaskCuBLAS cublasWrapper;
    const int Msize = 2;
    double matrix[Msize * Msize] = {1.0, 2.0, 3.0, 4.0};
    double* dev_ptr = nullptr;

    cublasWrapper.cublas_setmat(Msize, matrix, &dev_ptr);

    cublasWrapper.cublas_release(dev_ptr);

    std::cout << "C++ test completed successfully." << std::endl;
    return 0;
}